src/AzureRMTools.ts

```diff
@@ -321,6 +286,145 @@ export class AzureRMTools {
         }
     }
 
+    private logDeploymentTemplateErrors(document: vscode.TextDocument, deploymentTemplate: DeploymentTemplate): void {
+        deploymentTemplate.errors
+            .then((parseErrors: language.Issue[]) => {
+                const diagnostics: vscode.Diagnostic[] = [];
+
+                for (const error of parseErrors) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, error, vscode.DiagnosticSeverity.Error));
+                }
+
+                for (const warning of deploymentTemplate.warnings) {
+                    diagnostics.push(this.getVSCodeDiagnosticFromIssue(deploymentTemplate, warning, vscode.DiagnosticSeverity.Warning));
+                }
+
+                this._diagnosticsCollection.set(document.uri, diagnostics);
+            })
+            .catch((error: any) => {
+                this.logError("UnhandledError", error);
+            });
+    }
+
+    /**
+     * Update survey settings on disk and return the updated version
+     */
+    private updateSurveySettingsFromOpeningTemplate(): { surveySettings: SurveySettings, surveySettingsFilePath: string } {
+        const surveySettingsFilePath: string = path.join(os.homedir(), ".vscode", "azurerm", "surveyMetadata.json");
+        const surveySettings: SurveySettings = SurveySettings.fromFile(surveySettingsFilePath);
+
+        surveySettings.incrementDeploymentTemplatesOpenedOrCreated();
```

Comment:
Can the function be called incrementExistingTemplates? I'm not a fan of long names. :P